sdk/storage/azure-storage-common/src/xml_wrapper.cpp

```diff
@@ -396,9 +396,21 @@ namespace Azure { namespace Storage { namespace _internal {
 
   static void XmlGlobalInitialize() { static XmlGlobalInitializer globalInitializer; }
 
-  struct XmlReaderContext
+  XmlReader::XmlReader(XmlReader&& other) noexcept { *this = std::move(other); }
+  XmlReader& XmlReader::operator=(XmlReader&& other) noexcept
+  {
+    m_context = std::move(other.m_context);
+    m_context.reset();
```

Comment:
Why is `m_context` reset right after the assignment? Do you mean `other.m_context.reset()`?